modules/javafx.base/src/test/java/test/javafx/beans/value/ObservableValueFluentBindingsTest.java

```diff
@@ -0,0 +1,700 @@
+                            ReferenceAsserts.testIfNotStronglyReferenced(observableValue, () -> observableValue = null);
+                        }
+                    }
+                }
+            }
+
+            @Nested
+            class When_orElse_CalledReturns_ObservableValue_Which {
+                {
+                    observableValue = observableValue.orElse("Empty");
+                }
+
+                @Test
+                void shouldNotBeNull() {
+                    assertNotNull(observableValue);
+                }
+
+                @Test
+                void shouldNotBeStronglyReferenced() {
+                    ReferenceAsserts.testIfNotStronglyReferenced(observableValue, () -> observableValue = null);
+                }
+
+                @Nested
+                class WhenObserved {
+                    {
+                        startObserving(observableValue);
+                    }
+
+                    @Test
+                    void shouldApplyFlatMapThenMapOperation() {
+                        assertNothingIsObserved();
+
+                        property.set("Left");
+
+                        assertObserved("LEFT");
+
+                        property.set("Right");
+
+                        assertObserved("RIGHT");
+
+                        left.set("LEFT-LEFT");  // should have no effect as right branch is observed
+
+                        assertNothingIsObserved();
+
+                        right.set("RIGHT-RIGHT");
+
+                        assertObserved("RIGHT-RIGHT");
+
+                        right.set(null);
+
+                        assertObserved("Empty");
+
+                        property.set("Left");
+
+                        assertObserved("LEFT-LEFT");
+
+                        property.set(null);
+
+                        assertObserved("Empty");
+                    }
+
+                    @Test
+                    void shouldBeStronglyReferenced() {
+                        ReferenceAsserts.testIfStronglyReferenced(observableValue, () -> observableValue = null);
+                    }
+
+                    @Nested
+                    class AndWhenUnobserved {
+                        {
+                            stopObserving(observableValue);
+                        }
+
+                        @Test
+                        void shouldNoLongerBeCalled() {
+                            assertNothingIsObserved();
+
+                            property.set("Left");
+
+                            assertNothingIsObserved();
+                        }
+
+                        @Test
+                        void shouldNoLongerBeStronglyReferenced() {
+                            ReferenceAsserts.testIfNotStronglyReferenced(observableValue, () -> observableValue = null);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Ensures nothing has been observed.
```

Comment:
"Ensures nothing has been observed since the last check" or something like that because the values get cleared.

Reply:
Do you think it's not necessary?

Reply:
I missed this one during staging, added it the 2nd time.